src/plugins/intel_cpu/src/config.cpp

```diff
@@ -73,16 +68,14 @@ void Config::applyDebugCapsProperties() {
 void Config::readProperties(const ov::AnyMap& prop, const ModelType modelType) {
     const auto streamExecutorConfigKeys =
         streamExecutorConfig.get_property(ov::supported_properties.name()).as<std::vector<std::string>>();
-    const auto hintsConfigKeys = perfHintsConfig.SupportedKeys();
     for (const auto& kvp : prop) {
         const auto& key = kvp.first;
-        const auto& val = kvp.second.as<std::string>();
-        IE_SUPPRESS_DEPRECATED_START
+        const auto& val = kvp.second;
         if (streamExecutorConfigKeys.end() !=
             std::find(std::begin(streamExecutorConfigKeys), std::end(streamExecutorConfigKeys), key)) {
-            streamExecutorConfig.set_property(key, val);
+            streamExecutorConfig.set_property(key, val.as<std::string>());
```

Comment:
should we use `as<actual types>` instead of `std::string` everywhere?

Reply:
After checking the code, I think we cannot use `as<actual types>` because we still need support legacy properties(config/metric) which use std::string as property value.
We can change to be `val.as<actual_type>()` until all legacy properties has been removed.

Reply:
Update to `as<actual types>`.

Reply:
It seems there are some problem to cause `as<actual_type>()` failed:
1. python property map to C++ will use `int64 `for all `int `data, it cause `as<int32_t>/<uint32_t>` failed
       https://github.com/openvinotoolkit/openvino/blob/b1705e8bd3081dd5f8cf843a8fa7d0f7a545de60/src/bindings/python/src/pyopenvino/utils/utils.cpp#L301-L302
       I'm not sure whether it is a bug, but it does cause `any.as<int>()` failed.

2. Some non standard properties initialization, such as:
     https://github.com/openvinotoolkit/openvino/blob/b1705e8bd3081dd5f8cf843a8fa7d0f7a545de60/src/plugins/intel_cpu/tests/functional/behavior/ov_executable_network/properties.cpp#L118
    It it set `int` value to  `config[ov::num_streams.name()]`, but we use ` auto streams = value.as<ov::streams::Num>();` to convert it, so if failed.


For such questions, I think using std::string can avoid them. @ilya-lavrenov  do you have any suggestion?

Reply:
Considered above 2 problems, `non-int` data type will use `as<actual_type>`, and` int `data type still need `as<std::string>` to unify it.

Reply:
> python property map to C++ will use int64 for all int data, it cause as<int32_t>/<uint32_t> failed

maybe we need to ask to fix Python API to set correct values? If property has specific type, users must set values of those types

Reply:
ok, to keep `string` in this place since we passing values to another config